src/shared/components/query/GenesetsSelector.tsx

```diff
@@ -114,12 +119,36 @@ export default class GenesetsSelector extends QueryStoreComponent<GenesetsSelect
                         gsvaProfile={this.store.getFilteredProfiles("GENESET_SCORE")[0].molecularProfileId}
                         sampleListId={this.store.defaultSelectedSampleListId}
                         onSelect={map_geneset_selected => {
-                            this.store.applyGenesetsSelection(map_geneset_selected);
+                            this.store.applyGenesetSelection(map_geneset_selected);
                             this.store.showGenesetsHierarchyPopup = false;
                         }}
                     />
                     </Modal.Body>
                     </Modal>
+                    
+                    <Modal
+                        className={classNames('cbioportal-frontend',styles.GenesetsVolcanoSelectorWindow)}
+                        show={this.store.showGenesetsVolcanoPopup}
+                        onHide={() => this.store.showGenesetsVolcanoPopup = false }
+                        onShow={() => this.store.map_genesets_selected_volcano.replace(this.store.genesetIds.map(geneset => [geneset, true]))}
+                    >
+                    <Modal.Header closeButton>
+                    <Modal.Title>Select Gene Sets From Volcano Plot</Modal.Title>
+                    </Modal.Header>
+                    <Modal.Body>
+                    <GenesetsVolcanoSelector
+                        initialSelection={this.store.genesetIds}
+                        data={this.store.volcanoPlotTableData.result}
+                        plotData={this.store.volcanoPlotGraphData}
+                        maxY={this.store.maxYVolcanoPlot ? -(Math.log(this.store.maxYVolcanoPlot)/Math.log(10)) : undefined}
```

Comment:
i think this will show an empty chart.  you should check to see if the promise on which moxYVolcanoPlot depends isComplete

Reply:
No because it is the return data of get maxYVolcano plot: if the table data is not there, this function returns 'undefined', else returns the number.